### ISSUE TITLE
Wrong result with ILGPU 0.9.2

### ISSUE THREAD

**Issue by samy2** [m4rs-mt/ILGPU#331]
Hi,
This is quite strange. This test works with ILGPU 0.8.1.1, but fails with 0.9.2

        [Test]
        public void Test2()
        {
            using var context = new Context();
            using var gpu = new CudaAccelerator(context);
            var gpu_result1 = gpu.Allocate<int>(1);
            var gpu_result2 = gpu.Allocate<int>(1);
            var kernel = gpu.LoadKernel<ArrayView<int>,ArrayView<int>>(MyKernel2);
            kernel(gpu.DefaultStream, new KernelConfig(1, 1), gpu_result1, gpu_result2);
            var result1 = gpu_result1.GetAsArray(gpu.DefaultStream);
            var result2 = gpu_result2.GetAsArray(gpu.DefaultStream);
            Assert.AreEqual(0, result1[0]);     // OK
            Assert.AreEqual(1, result2[0]);     // KO : expected 1, but was 6
        }

        private static void MyKernel2(ArrayView<int> gpu_result1, ArrayView<int> gpu_result2)
        {
            var threadIdx = ILGPU.Group.IdxX;
            var warpId = threadIdx >> 5;
            gpu_result1[threadIdx] = warpId;
            gpu_result2[threadIdx] = warpId + 1;
        }

**Comment by m4rs-mt** [comment]
@samy2 Thank your for your issue report 🔢 Unfortunately, I am not able to reproduce your issue using the sample provided (neither with v0.9.2, nor with the latest master branch). Can you provide more information?

**Comment by samy2** [comment]
@m4rs-mt 
Ok, that's strange...
My project is targeting .Net framework 4.8. I'm using VS 2019 version 16.8.2.
I'm using the official ILGPU 0.9.2 package from Nuget.org :
`<package id="ILGPU" version="0.9.2" targetFramework="net48" />`
The project works well with ILGPU 0.8.1.1

**Comment by samy2** [comment]
I will try to investigate more, and reproduce the problem in an empty project.

**Comment by m4rs-mt** [comment]
@samy2 Hm... I just tried v0.9.2 from nuget; without any success at the moment 😢 I also checked the assembler code, and it looks pretty good to me:
```asm
	mov.b32	%r1, %tid.x;
*	shr.s32	%r2, %r1, 5;
	mul.wide.u32	%rd6, %r1, 4;
	add.u64	%rd5, %rd1, %rd6;
	st.b32	[%rd5], %r2;
	mul.wide.u32	%rd7, %r1, 4;
	add.u64	%rd6, %rd3, %rd7;
*	add.s32	%r3, %r2, 1;
	st.b32	[%rd6], %r3;
```
However, this could be a concurrency problem 🤔 This the program always yield the same "wrong" result?

**Comment by samy2** [comment]
Ok @m4rs-mt, 
yes always the same wrong result.
And if I change 
`var warpId = threadIdx >> 5;`
to 
`var warpId = threadIdx >> 6;`
the wrong result is 7 (6+1) !

But I just created an empty project from zero, and the problem do not occure any more !
I will investigate more, thank you for your help !

**Comment by m4rs-mt** [comment]
@samy2 Hm... I guess we should try to investigate this in more detail 🤓😟

**Comment by samy2** [comment]
I really don't know what happened, but after removing and reinstalling the ILGPU 0.9.2 package, the problem does not occurs any more ! Sorry for that buggy bug report.
That's really strange, because ILGPU seemed to work : most of my kernels were working ! And the "bug" was 100% reproductible...
@m4rs-mt , Thank you again for your time, and for this wonderful project !

**Comment by MoFtZ** [comment]
@samy2 is this related to the ILGPU v0.9.2-beta1? There was a bug with constant propagation in the beta.

Maybe when updating from beta to final version of v0.9.2 is having issues.

**Comment by samy2** [comment]
No, I didn't install v0.9.2-beta1.
But I remember I build a patched version of v0.9.1 (I created a local version v0.9.1.1), to test the correction of #292.
I guess I messed up something in the process.

**Comment by m4rs-mt** [comment]
I will close this issue for now. If the problem persists or reoccurs, we can reopen the issue or create a new one.
